process/sync/metablock.go

```diff
@@ -369,6 +372,11 @@ func (boot *MetaBootstrap) StopSync() {
 func (boot *MetaBootstrap) syncBlocks() {
 	for {
 		time.Sleep(sleepTime)
+
+		if !boot.networkWatcher.IsConnectedToTheNetwork() {
+			continue
```

Comment:
add a test for this. All the tests in the test file return true for `IsConnectedToTheNetwork`

Reply:
Will be done in this PR at a latter time. It depends on a pending refactoring.